sublime-package.json

```diff
@@ -26,6 +26,41 @@
               "type": "string",
               "deprecationMessage": "Use the \"selector\" key instead."
             },
+            "FileWatcher": {
```

Comment:
should we expose the FileWatcher in the settings and let the user (miss)configure it?

Reply:
You mean global (LSP) settings? I don't think that would make sense. It's something specific to a language server.

Reply:
I read the spec a bit more:

> The watched files notification is sent from the client to the server when the client detects changes to files and folders watched by the language client (note although the name suggest that only file events are sent it is about file system events which include folders as well). 

Who defines the glob patterns?
The server or the client.

I thought it was the server, now I think it is the client who defines `glob`s and `kind`s.

Reply:
It's the server that wants to be notified about specific file events. For example vetur wants to know about changes in *.js files as it doesn't run in JS files but changes in JS files might affect the *.vue files. Or eslint server wants to know about changes in .eslintrc.* configuration file(s).

A generic client doesn't know what the server is interested in watching.

With dynamic registrations being available, the server triggers those to register the watchers.

Before dynamic registrations there was no way for server to register a watcher since there is no server capability for it or anything like that so before it worked because the client (vscode extension) had hardcoded patterns for the server that it was running.

Reply:
But now that dynamic capabilities exist for a while, is it still needed to have static glob patterns?

Do you have an example of a server that relies on the client knowing the correct glob patterns?

Reply:
vetur and eslint, as mentioned, still rely on the watcher being registered by the client (the extension in case of vscode).